File: Java/SAMPLE2.java
// Distributed with a free-will license.
// Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
// MCP23008
// This code is designed to work with the MCP23008_I2CR8G5LE_10A I2C relay controller available from ControlEverything.com.
// https://www.controleverything.com/content/Relay-Controller?sku=MCP23008_I2CR8G5LE_10A#tabs-0-product_tabset-2

import com.pi4j.io.i2c.I2CBus;
import com.pi4j.io.i2c.I2CDevice;
import com.pi4j.io.i2c.I2CFactory;
import java.io.IOException;

public class SAMPLE2
{
	public static void main(String args[]) throws Exception
	{
		// Create I2C bus
		I2CBus bus = I2CFactory.getInstance(I2CBus.BUS_1);
		// Get I2C device, MCP23008 I2C address is 0x20(32)
		I2CDevice device = bus.getDevice(0x20);
		
		// Configure all pins as OUTPUT
		device.write(0x00, (byte)0x00);
		
		// Turn all relays ON, one by one
		System.out.println("Turning all relays ON");
		byte data = 0x01;
		for (int i = 0; i < 8; i++)
		{
			device.write(0x09, (byte)data);
			data = (byte)(data << 1);
			data += 1;
			Thread.sleep(1000);
		}
		
		// Turn all relays OFF, one by one
		System.out.println("Turning all relays OFF");
		data = (byte)0xFE;
		for (int i = 0; i < 8; i++)
		{
			device.write(0x09, (byte)data);
			data = (byte)(data << 1);
			Thread.sleep(1000);
		}
	}
}

File: Python/Sample2.py
# Distributed with a free-will license.
# Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
# MCP23008
# This code is designed to work with the MCP23008_I2CR8G5LE_10A I2C relay controller available from ControlEverything.com.
# https://www.controleverything.com/content/Relay-Controller?sku=MCP23008_I2CR8G5LE_10A#tabs-0-product_tabset-2

import smbus
import time

# Get I2C bus
bus = smbus.SMBus(1)

# MCP23008 address, 0x20(32)
# Select IODIR register, 0x00(00)
#		0x00(00)	All pins are configured as outputs
bus.write_byte_data(0x20,0x00,0x00)

data = 0x01
for MyData in range(0,8):
# MCP23008 address, 0x20(32)
# Select GPIO register, 0x09(9)
	bus.write_byte_data(0x20,0x09,data)
	print "GPIO Pin %d State is HIGH" %MyData
	data = data << 1
	data = data + 1
	time.sleep(1)

data = 0xFE
for MyData in range(0,8):
# MCP23008 address, 0x20(32)
# Select GPIO register, 0x09(9)
	bus.write_byte_data(0x20,0x09,data)
	print "GPIO Pin %d State is LOW" %MyData
	data = data << 1
	time.sleep(1)
